src/main/java/dk/kb/present/copyright/CopyrightAccessExtractor.java

```diff
@@ -0,0 +1,364 @@
+ * This should be the newest version and is stil not finished. (materiel type logic missing)
+ * The version used to implement this class can be found in the /doc folder in the project
+ * 
+ */
+public class CopyrightAccessExtractor {
+
+    private static final Logger log = LoggerFactory.getLogger(CopyrightAccessExtractor.class);
+
+    public static CopyrightAccessDto extractCopyrightFields(String xml) throws Exception {
+
+        CopyrightAccessDto copyrightDto= new CopyrightAccessDto();
+
+        Document document = createDocFromXml(xml);
+        
+        copyrightDto.setMaterialeType(getMaterialType(document));
+        copyrightDto.setSkabelsesAar(getSkabelsesAar(document));
+        
+        
+        //TEMORARY SOLUTION TO SET IMAGE LINK!
+        copyrightDto.setImageUrl(getImageLink(document));
+        
+        NodeList nList = document.getElementsByTagName("mets:rightsMD");
+        if (nList.getLength()==0) {
+            log.info("No rightsMD found");
+            copyrightDto.setAccessConditionsList(new ArrayList<AccessCondition>()); //Set empty list
+            return copyrightDto;
+        }
+        
+        String version=nList.item(0).getAttributes().getNamedItem("ID").getNodeValue();
+        if (!"ModsRights1".equals(version)) { //Failed hard if we are not parsing the implemented version
+            throw new InvalidArgumentServiceException("'Mets:rightsMD' Version not supported:"+version);          
+        }
+
+        Element rightsMD = (Element) nList.item(0);
+
+        NodeList accessConditions = rightsMD.getElementsByTagName("mods:accessCondition");        
+        ArrayList<AccessCondition> accessConditionList = buildAccessCondition(accessConditions);
+        copyrightDto.setAccessConditionsList(accessConditionList);        
+        
+        Integer lastDeathYearForPerson = getLastDeathYearForPerson(accessConditionList);
+        if (lastDeathYearForPerson !=  null) {
+          copyrightDto.setOphavsPersonDoedsAar(lastDeathYearForPerson);;
+        }
+        
+        return  copyrightDto;
+    }
+
+
+    /*
+   <mods:accessCondition type="use and reproduction" displayLabel="Restricted ">Ejermærke</mods:accessCondition>    
+     <cdl:copyright publication.status="unknown" copyright.status="copyrighted" xsi:schemaLocation="http://www.cdlib.org/inside/diglib/copyrightMD /usr/local/ginnungagap/current/script/xsd/copyright-md.xsd">
+       <cdl:creator>
+         <cdl:creator.person>
+           <cdl:name>Clemens, Johann Friderich</cdl:name>
+           <cdl:year.birth>1748-11-29</cdl:year.birth>
+           <cdl:year.death>1831-11-5</cdl:year.death>
+         </cdl:creator.person>
+       </cdl:creator>
+     </cdl:copyright>   
+     */
+    public static ArrayList<AccessCondition> buildAccessCondition(  NodeList accessCondititions){       
```

Comment:
`buildAccessConditions`, not `buildAccessCondition`. But more importantly: what does the method do? Is it plain DTO generation (I guess so?) or does it perform business login? JavaDoc!

Reply:
fixed

Reply:
You don't need to defend the decision in the JavaDoc (half of the text). the first part "Maps the accessConditions tags (there can be multiple) to a java DTO. There is much cross-logic involved" is fine.